src/dali_executor/dali_pipeline.h

```diff
@@ -116,13 +122,30 @@ class DaliPipeline {
   void SetInput(const void* ptr, const char* name, device_type_t source_device,
                 dali_data_type_t data_type, TensorListShape<> input_shape);
 
+  void SetInput(const IDescr& io_descr);
+
   void PutOutput(void* destination, int output_idx, device_type_t destination_device);
 
+  /**
+   * @brief Wait for all output copies.
+   *
+   * This should be always called after copying all of the pipeline outputs.
+   */
+  void SyncOutputStream();
+
   void Reset() {
     ReleasePipeline();
     CreatePipeline();
   }
 
+  int DeviceId() {
+    return device_id_;
+  }
+
+  int NumThreadsArg() {
```

Comment:
It's because it's not the actual number of threads used by the pipeline but the argument passed to the constructor (it might be equal -1). I have no access to the actual number of threads used by the pipeline.